### ISSUE TITLE
Incorrect message when branch is not recognized as release branch

### ISSUE THREAD

**Issue by gr2m** [semantic-release/semantic-release#1443]
## Current behavior

I get this error message when I was trying to publish from a `v1.x` branch (instead of `1.x`). The message says

> This test run was triggered on the branch v1.x, while semantic-release is configured to only publish from master, therefore a new version won’t be published.

Althoug in the same logs above it says

> 2020-02-02T00:59:10.827Z semantic-release:config options values: {
>   branches: [
>     '+([0-9])?(.{+([0-9]),x}).x',
>     'master',
>     'next',
>     'next-major',
>     { name: 'beta', prerelease: true },
>     { name: 'alpha', prerelease: true }
>   ],

## Expected behavior

It should list the branches it is configured to publish from, not just say "master"

## Environment

- **semantic-release** version: 17.0.2
- CI environment: GitHub Actions
- Plugins used: _defaults_
- **semantic-release** configuration: _defaults_
- CI logs: https://github.com/octokit/plugin-paginate-rest.js/runs/421313433?check_suite_focus=true

**Comment by pvdlg** [comment]
That's the intended behavior (as in the code is doing what I wanted, but not necessarily the best thing we could be doing). The error message lists the branches present on your remote that matches what's defined in your config. In your case it's only `master` because you don't have a `next`, `next-major`, `beta`, `alpha` or any branches matching the pattern `+([0-9])?(.{+([0-9]),x}).x` in your remote.

What message would you like to see in such case? Do you have an example of the phrasing you'd like?

**Comment by gr2m** [comment]
Ah I see. 

I would have expected that `semantic-release` would log the `branches` that are set by default. In my case, if the error message would have been

> This test run was triggered on the branch v1.x, while semantic-release is configured to publish from master, next, next-major, alpha, beta, or /+([0-9])?(.{+([0-9]),x}).x/. Therefore a new version won’t be published.

I would have realized that the problem was the `v` in my branch. With the current message I thought that I somehow had only "master" configured as publish branch, but I couldn't figure it where.

**Comment by pvdlg** [comment]
Maybe we should go a bit further an include in this message both the configuration and the actual matching branches found in the remote. That would avoid confusion.

**Comment by gr2m** [comment]
Yeah, the clearer, the better.

**Comment by adrianschmidt** [comment]
> Maybe we should go a bit further an include in this message both the configuration and the actual matching branches found in the remote. That would avoid confusion.

Yes, that would be great. I've been stumped by this as well, as I've been trying to update from v15 to v16+.

Your explanation makes sense, but I wasn't able to understand what the error was trying to say without the explanation 🙂

**Comment by jpedroh** [comment]
Is this bug still a thing? Looking at index file
https://github.com/semantic-release/semantic-release/blob/971a5e0d16f1a32e117e9ce382a1618c8256d0d9/index.js#L60-L67
It seems to have been fixed. If it hasn't though, I would be interested in working in a PR to address this

**Comment by EvanCarroll** [comment]
> That's the intended behavior (as in the code is doing what I wanted, but not necessarily the best thing we could be doing). The error message lists the branches present on your remote that matches what's defined in your config. In your case it's only master because you don't have a next, next-major, beta, alpha or any branches matching the pattern +([0-9])?(.{+([0-9]),x}).x in your remote.
> What message would you like to see in such case? Do you have an example of the phrasing you'd like?

I would like it to just work as in if you fork this into another repo and publish to an alpha branch I would like semantic-release should just run (bump version, generate change log, push to fork). What's the advantage of bailing out at all?

I also found this confusing #2150

**Comment by coyaSONG** [comment]
I’d like to work on this. I plan to update the non-release-branch message to show both the configured branch entries and the matching branches found on the remote, with a focused integration regression. I’m using Codex assistance and will keep the change scoped to this issue.
